src/ucs/sys/sock.c

```diff
@@ -850,6 +850,19 @@ ucs_status_t ucs_sockaddr_copy(struct sockaddr *dst_addr,
     return UCS_OK;
 }
 
+void ucp_sockaddr_copy_always(struct sockaddr *dst_addr,
+                              const struct sockaddr *src_addr)
+{
+    ucs_status_t status;
+
+    if (!ucs_sockaddr_is_known_af(src_addr)) {
+        memset(dst_addr, 0, sizeof(*dst_addr));
+    } else {
+        status = ucs_sockaddr_copy(dst_addr, src_addr);
+        ucs_assert_always(status == UCS_OK);
+    }
```

Comment:
if AF not known ucs_sockaddr_copy would fail anyway

Reply:
@yosefe Yes, so ucs_sockaddr_copy is called only when AF is known. What we should change here?

Reply:
@yosefe Or do you mean no need to check the condition ucs_sockaddr_is_known_af(src_addr), just do ucs_sockaddr_copy() anyway?

Reply:
yes, no need to define ucp_sockaddr_copy_always() function
also it's ucs function so should have been uc**s**_sockaddr_copy_always

Reply:
@yosefe Done